api-lib/sendSocialMessage.ts

```diff
@@ -176,16 +174,105 @@ export async function sendSocialMessage({
       chat_id: channelId,
       text: msg,
     };
-    const res = await fetch(`${TELEGRAM_BOT_BASE_URL}/sendMessage`, {
-      method: 'POST',
+
+    const updateTelegramBot = fetchUrl({
+      url: `${TELEGRAM_BOT_BASE_URL}/sendMessage`,
       body: JSON.stringify(telegramBotPost),
+      label: 'telegram-bot',
+      notifyOrg,
+      circleId,
+      channelId,
+    });
+    requests.push(updateTelegramBot);
+  }
+
+  const responses = await Promise.allSettled(requests);
+
+  const errors = responses.filter(isRejected);
+
+  if (errors.length > 0) {
+    const errorMessages = errors.map(err => {
+      if (err.reason instanceof Error) {
+        return err.reason.stack;
+      } else {
+        return String(err.reason);
+      }
+    });
+    console.error(`Error sending social messages: ${errorMessages.join('\n')}`);
+    throw new Error(errorMessages.join('\n'));
+  }
+}
+
+const isRejected = (
+  response: PromiseSettledResult<unknown>
+): response is PromiseRejectedResult => response.status === 'rejected';
+
+const fetchUrl = async ({
+  url,
+  headers,
+  body,
+  label,
+  notifyOrg,
+  channelId,
+  circleId,
+}: {
+  url: RequestInfo;
+  headers?: HeaderInit;
+  body: BodyInit;
+  label: string;
+  notifyOrg: boolean;
+  channelId?: string;
+  circleId: number;
+}) => {
+  try {
+    const updateWebhook = await fetch(url, {
```

Comment:
this shouldn't be called `updateWebhook` anymore because it's general-purpose now; how about just `response` or `resp`?